config.go

```diff
@@ -160,23 +120,74 @@ func makeCfg(config []Config) (cfg Config) {
 			cfg.AuthScheme = "Bearer"
 		}
 	}
-	if cfg.KeyRefreshTimeout == nil {
-		cfg.KeyRefreshTimeout = &defaultKeyRefreshTimeout
-	}
 
 	if cfg.KeyFunc == nil {
-		if len(cfg.KeySetURLs) > 0 {
-			jwks := &KeySet{
-				Config: &cfg,
+		if len(cfg.SigningKeys) > 0 || len(cfg.JWKSetURLs) > 0 {
+			var givenKeys map[string]keyfunc.GivenKey
+			if cfg.SigningKeys != nil {
+				givenKeys = make(map[string]keyfunc.GivenKey, len(cfg.SigningKeys))
+				for kid, key := range cfg.SigningKeys {
+					givenKeys[kid] = keyfunc.NewGivenCustom(key, keyfunc.GivenKeyOptions{}) // TODO User supplied alg?
```

Comment:
The given keys in this project's `Config` data structure have the opportunity to contain a user supplied algorithm. This can be passed to the `keyfunc` package and those values will be used in JWT cryptographic verification.

Is that a desired change? I would likely change the type of the value in the `SigningKeys` `map[string]interface{}` to a non-`interface{}` value.

Reply:
Agree with you

Reply:
I've updated how users can specify which JWT algorithm is requried.